File: Array/1287. Element Appearing More Than 25% In Sorted Array.cpp
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {
        int n=arr.size();
        int perc=(int)(0.25*n);
        unordered_map<int,int> mp;
        for(auto n:arr){
            mp[n]++;
        }
        for(auto p:mp){
            if(p.second>perc){
                return p.first;
            }
        }
        return -1;
    }
};

File: Array/1287. Element Appearing More Than 25% In Sorted Array.java
class Solution {
    public int findSpecialInteger(int[] arr) {
        HashMap<Integer,Integer> mp=new HashMap<>();
        int perct=(int)(0.25*arr.length);
        for(int n:arr){
            mp.put(n,mp.getOrDefault(n,0)+1);
        }
        for(Map.Entry<Integer,Integer> entry:mp.entrySet()){
            if(entry.getValue()>perct){
                return entry.getKey();
            }
        }
        return -1;
    }
}
